internal/handlers/handler_user_info.go

```diff
@@ -74,6 +74,24 @@ func loadInfo(username string, storageProvider storage.Provider, userInfo *UserI
 		userInfo.HasTOTP = true
 	}()
 
+	go func() {
+		defer wg.Done()
+
+		_, _, err := storageProvider.LoadPreferredDuoDevice(username)
+		if err != nil {
+			if err == storage.ErrNoDuoDevice {
+				return
+			}
+
+			errors = append(errors, err)
+			logger.Error(err)
+
+			return
+		}
+
+		userInfo.HasDuo = true
+	}()
```

Comment:
Would this cause issues for people already using duo? Guessing they'll just have to select a device.

Reply:
If they only have a single device then the Duo PreAuth will pick it up and set it as the preferred device.

If a user has multiple they will need to select a device as you suspected.